### PR TITLE
fix: discard pending changes

### PR DESCRIPTION
**Pull request by Shurtu-gal**
Close: #6549

## PR Details

- The PR fixes the problem of errors when discarding pending changes of newly built service.

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
good work
please see my questions

File: packages/amplication-prisma-db/prisma/schema.prisma [hunk 1]
```diff
@@ -297,7 +297,7 @@ model EntityField {
   description      String
   position         Int?
   unique           Boolean                 @default(false)
-  entityVersion    EntityVersion           @relation(fields: [entityVersionId], references: [id], onDelete: Cascade)
+  entityVersion    EntityVersion?          @relation(fields: [entityVersionId], references: [id], onDelete: Cascade)
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply packages/amplication-prisma-db/prisma/schema.prisma:300]: Oh, sorry it was left out by mistake when I was trying various solutions. Will remove it.

**Review by yuval-hazaz — APPROVED** [review]

File: packages/amplication-server/src/core/serviceSettings/serviceSettings.service.ts [hunk 2]
```diff
@@ -61,6 +60,8 @@ export class ServiceSettingsService {
       );
     }
 
+    serviceSettings.serverSettings.generateServer = true;
+
```
Comment (abrl91) [inline packages/amplication-server/src/core/serviceSettings/serviceSettings.service.ts:64]: you moved this line without moving its comment (line 54)

Comment (Shurtu-gal) [reply packages/amplication-server/src/core/serviceSettings/serviceSettings.service.ts:64]: Okay, I will make a new PR.

Comment (Shurtu-gal) [reply packages/amplication-server/src/core/serviceSettings/serviceSettings.service.ts:64]: @abrl91 The PR :- https://github.com/amplication/amplication/pull/7807

**Pull request merged**